### DISCUSSION TITLE
Comma below footer on Gatsby site

### DISCUSSION THREAD

**Discussion by kim-lindgren** [gatsbyjs/gatsby#35180] [Help] [answered]
I have a comma below my footer on a Gatsby site and I cannot find its source. It is at the end of the wrapper div with id "___gatsby". I have searched through countless of files to no avail. Going through all previous Git versions to find the change would be very burdensome at this point. Perhaps someone has an idea, based on the DOM structure below, where to best search for the comma?

Thanks.

![](https://i.stack.imgur.com/wmVvT.png)

**Comment by matthewhoth** [comment]
In Visual Code do a search for "," and start scanning. 

It will tell you there are thousands of results, but most of those can be ignored. Simply scroll and look for a single comma.

", " might even find the result instantly since most commas don't have a space after them.

**Reply by kim-lindgren** [reply]
I should have clarified that I have already searched in this manner. I have not looked through all thousands of results but tried to go through all files that could be relevant to no avail.

The thing is that this comma exists outside of the #gatsby-focus-wrapper where all my components are. Now I have searched through the components anyway but it isn't there. Based on the DOM structure of the site, would it not be possible to tell where the comma might come from? It is within the #___gatsby div but outside of its child divs. What causes any character to be inserted here? If someone understands how Gatsby compiles a bit better than me, then perhaps they could make a qualified guess about where to look... or so I hope. I would personally look for an index.html file but this does not exist and I am not aware of any Gatsby equivalent.

**Reply by kim-lindgren** [reply]
I have gone through all js-files in /components and /pages, even though it doesn't make sense that the comma would be there. But what other files could cause anything to render?

**Reply by matthewhoth** [reply]
Yeah someone might be able to tell you just from its location.

If you've got the whole code I don't mind taking a look and finding it for ya.

**Comment by kim-lindgren** [comment, marked as answer]
I found it.

It turns out that a rogue character in gatsby-browser.js, in this case a comma after a closing tag, will be added in this way to the end of the #___gatsby div. In this case I overlooked the comma even through search because the comma was at the end of a long line of code and so did not appear visually in the search results list.

Thanks anyway!
